lib/api/BackbeatAPI.js

```diff
@@ -428,23 +398,143 @@ class BackbeatAPI {
-        const { failedCRR } = redisKeys;
-        const pattern = `${failedCRR}:${bucket}:${key}:${versionId}:*`;
-        return this._scanAllKeys(pattern, 0, [], (err, cursor, keys) =>
-            this._getFailedCRRResponse(cursor, keys, cb));
+        const member = getSortedSetMember(bucket, key, versionId);
+        return this._getEntriesAcrossSite(member, (err, entries) =>
+            this._getFailedCRRResponse(undefined, entries, cb));
     }
 
     /**
-     * Get all CRR operations that have failed.
+     * Get all CRR operations that have failed by site.
      * @param {Object} details - The route details
      * @param {Function} cb - The callback to call
      * @return {undefined}
      */
-    getAllFailedCRR(details, cb) {
-        const marker = Number.parseInt(details.marker, 10) || 0;
-        const pattern = `${redisKeys.failedCRR}:*`;
-        return this._scanAllKeys(pattern, marker, [], (err, cursor, keys) =>
-            this._getFailedCRRResponse(cursor, keys, cb));
+    getSiteFailedCRR(details, cb) {
+        const { marker, sitename } = details;
+        const score = Number.parseInt(marker, 10) || 0;
+        const keys = this._getSortedSetKeys(sitename);
+        return this._getEntriesBySite(score, sitename, keys,
+            (err, nextMarker, entries) =>
+                this._getFailedCRRResponse(nextMarker, entries, cb));
+    }
+
+    /**
+     * Get all entries across all site sorted sets.
+     * @param {String} member - The member in the sorted set to find
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    _getEntriesAcrossSite(member, cb) {
+        const entries = [];
+        return async.eachLimit(this._validSites, 10, (site, next) => {
+            const keys = this._getSortedSetKeys(site);
+            return async.eachLimit(keys, 10, (key, done) => {
+                this._redisClient.zscore(key, member, (err, res) => {
+                    if (err) {
+                        return done(err);
+                    }
+                    if (res !== null) {
+                        const entry = new ObjectFailureEntry(member, site);
+                        entries.push(entry);
+                    }
+                    return done();
+                });
+            }, next);
+        }, err => {
+            if (err) {
+                return cb(err);
+            }
+            return cb(null, entries);
+        });
+    }
+
+    /**
+     * Use the current score to find all sorted sets that are >= to the score.
+     * @param {Number} score - The score to use for reduction
+     * @param {Array} keys - All the possible site keys
+     * @return {Array} - The reduced ke
+     */
+    _reduceSortedSetKeys(score, keys) {
+        const date = new Date(score);
+        const startHour = this._statsClient.normalizeTimestampByHour(date);
+        const index = keys.findIndex(key => key.includes(startHour));
```

Comment:
I agree, this could be cleaned up. It could happen if the marker the user gives is a timestamp they've created. 

For now, it looks harmless, as `-1` will get the last sorted set key. The response will be an empty array because that sorted set wouldn't contain a member with the given marker as its score.